panic when unspecifying range of graph widget

```
Stack trace from the terraform-provider-mackerel_v0.3.0 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 70 [running]:
github.com/mackerelio-labs/terraform-provider-mackerel/mackerel.expandDashboardRange({0x101b67390?, 0x140000a8360?, 0x10144390a?})
	github.com/mackerelio-labs/terraform-provider-mackerel/mackerel/resource_mackerel_dashboard.go:522 +0x5bc
github.com/mackerelio-labs/terraform-provider-mackerel/mackerel.expandDashboardWidgets(0x1176ba15c7c?)
	github.com/mackerelio-labs/terraform-provider-mackerel/mackerel/resource_mackerel_dashboard.go:389 +0x2e10
github.com/mackerelio-labs/terraform-provider-mackerel/mackerel.expandDashboard(0x0?)
	github.com/mackerelio-labs/terraform-provider-mackerel/mackerel/resource_mackerel_dashboard.go:346 +0xbc
github.com/mackerelio-labs/terraform-provider-mackerel/mackerel.resourceMackerelDashboardUpdate({0x1016d59a8, 0x14000721800}, 0x140008d0000, {0x1016c9ae0?, 0x14000720420?})
	github.com/mackerelio-labs/terraform-provider-mackerel/mackerel/resource_mackerel_dashboard.go:324 +0x100
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x140002bdc00, {0x1016d5938, 0x1400047ae00}, 0x24?, {0x1016c9ae0, 0x14000720420})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:364 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x140002bdc00, {0x1016d5938, 0x1400047ae00}, 0x140004b69a0, 0x140005bbde0, {0x1016c9ae0, 0x14000720420})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:464 +0x480
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000195338, {0x1016d5938, 0x1400047ae00}, 0x140007c4000)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:955 +0x868
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0x140001c57c0, {0x1016d59e0?, 0x1400022db60?}, 0x140008c4000?)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0x64
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x10169e0a0?, 0x140001c57c0}, {0x1016d59e0, 0x1400022db60}, 0x140007205a0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002a2c40, {0x1016d9398, 0x140000036c0}, 0x140002219e0, 0x140003f50e0, 0x101b28ce0, 0x0)
	google.golang.org/grpc@v1.46.2/server.go:1283 +0xb50
google.golang.org/grpc.(*Server).handleStream(0x140002a2c40, {0x1016d9398, 0x140000036c0}, 0x140002219e0, 0x0)
	google.golang.org/grpc@v1.46.2/server.go:1620 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.46.2/server.go:922 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.46.2/server.go:920 +0x290

Error: The terraform-provider-mackerel_v0.3.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/mackerel/resource_mackerel_dashboard.go
+++ b/mackerel/resource_mackerel_dashboard.go
@@ -354,12 +354,16 @@ func expandDashboardWidgets(d *schema.ResourceData) []mackerel.Widget {
 		graphs := d.Get("graph").([]interface{})
 		for _, graph := range graphs {
 			g := graph.(map[string]interface{})
+			var r mackerel.Range
+			if v, ok := g["range"].([]interface{}); ok && len(v) > 0 {
+				r = expandDashboardRange(v)
+			}
 			if len(g["host"].([]interface{})) > 0 {
 				widgets = append(widgets, mackerel.Widget{
 					Type:   "graph",
 					Title:  g["title"].(string),
 					Graph:  expandDashboardGraphHost(g["host"].([]interface{})),
-					Range:  expandDashboardRange(g["range"].([]interface{})),
+					Range:  r,
 					Layout: expandDashboardLayout(g["layout"].([]interface{})[0].(map[string]interface{})),
 				})
 			}
@@ -368,7 +372,7 @@ func expandDashboardWidgets(d *schema.ResourceData) []mackerel.Widget {
 					Type:   "graph",
 					Title:  g["title"].(string),
 					Graph:  expandDashboardGraphRole(g["role"].([]interface{})),
-					Range:  expandDashboardRange(g["range"].([]interface{})),
+					Range:  r,
 					Layout: expandDashboardLayout(g["layout"].([]interface{})[0].(map[string]interface{})),
 				})
 			}
@@ -377,7 +381,7 @@ func expandDashboardWidgets(d *schema.ResourceData) []mackerel.Widget {
 					Type:   "graph",
 					Title:  g["title"].(string),
 					Graph:  expandDashboardGraphService(g["service"].([]interface{})),
-					Range:  expandDashboardRange(g["range"].([]interface{})),
+					Range:  r,
 					Layout: expandDashboardLayout(g["layout"].([]interface{})[0].(map[string]interface{})),
 				})
 			}
@@ -386,7 +390,7 @@ func expandDashboardWidgets(d *schema.ResourceData) []mackerel.Widget {
 					Type:   "graph",
 					Title:  g["title"].(string),
 					Graph:  expandDashboardGraphExpression(g["expression"].([]interface{})),
-					Range:  expandDashboardRange(g["range"].([]interface{})),
+					Range:  r,
 					Layout: expandDashboardLayout(g["layout"].([]interface{})[0].(map[string]interface{})),
 				})
 			}

